play.py

```diff
@@ -0,0 +1,21 @@
+import player
+import game
+import argparse
+
+parser = argparse.ArgumentParser(description="setup tictactoe game")
+parser.add_argument('b', nargs='?', default=3, type=int, help="board size")
+parser.add_argument('p1', nargs='?', default="human", type=str, help="player2 human or computer")
+parser.add_argument('pi1', nargs='?', default="X", type=str, help="player1 piece shape")
+parser.add_argument('p2', nargs='?', default="computer", type=str, help="player2 human or computer")
+parser.add_argument('pi2', nargs='?', default="O", type=str, help="player2 piece shape")
+
+args = parser.parse_args()
+
+player1 = player.Player(args.p1, args.pi1)
+player2 = player.Player(args.p2, args.pi2)
+player1.opponent = player2
+player2.opponent = player1
+game1 = game.Game(player1, player2, args.b)
+
+if __name__ == '__main__':
+    game1.main()
```

Comment:
Here's a little refactor here that makes your CLI a lot easier to test by wrapping the parser in a function:
```
# file: __main__.py

# create a dictionary of all the options to pass to the parser
CLI_ARGS = {
   '-b': {
      'default': 3,
      'required': False,
   }, 
}

def build_parser(arg_dict=CLI_ARGS, **kwargs):
   parser = argparse.ArgumentParser(**kwargs)
   for (flag, options) in arg_dict.items():
       parser.add_argument(flag, **options)
   return parser

def main(args=sys.argv):
   parser = build_parser()    
   arg_context = parser.parse_args(args)
   game = Game(...) # do stuff with arg_context
   return game.main(...)

if __name__ == "__main__":
  sys.exit(main())
```

There's a number of nifty advantages here:
- You can now write tests for `build_parser` with different args
- The whole program `main` can be tested with arbitrary input
- The CLI behaviour now passes the result of main as an *exit code* so that the calling program/user is able to know whether the program terminated normally or not with `$?` in the shell (https://www.gnu.org/software/bash/manual/html_node/Exit-Status.html)